src/test/scala/firrtlTests/InferResetsSpec.scala

```diff
@@ -410,5 +405,90 @@ class InferResetsSpec extends FirrtlFlatSpec {
     result should containTree { case Port(_, "childReset", Input, BoolType) => true }
     result should containTree { case Port(_, "childReset", Input, AsyncResetType) => true }
   }
+
+  it should "infer based on what a component *drives* not just what drives it" in {
+    val result = compile(s"""
+      |circuit top :
+      |  module top :
+      |    input in : AsyncReset
+      |    output out : Reset
+      |    wire w : Reset
+      |    w is invalid
+      |    out <= w
+      |    out <= in
+      |""".stripMargin)
+    result should containTree { case DefWire(_, "w", AsyncResetType) => true }
+  }
+
+  it should "infer from connections, ignoring \"winning\" invalidation" in {
```

Comment:
Maybe reword this as "ignoring the fact that the invalidation wins"?

Reply:
yeah that's better